R/restore_labels.R

```diff
@@ -59,7 +59,9 @@ restore_labels <- function(x, newLabels,
 
   # Ensure class consistency
   if (!inherits(x, "safeframe")) {
-    class(x) <- c("safeframe", class(x))
+    current_classes <- class(x)
+    df_index <- which(current_classes == "data.frame")
+    class(x) <- append(current_classes, "safeframe", after = df_index - 1)
```

Comment:
As shown in the proposed new test, this will come short in complex cases where the safeframe is a subclass of a data.frame (e.g., a tibble).

I think the only solution is to restore the class in the function where it's dropped (i.e., the subsetting functions) and to restore a copy of the original class